src/ReactionService.Business/Commands/ReactionsGroup/CreateReactionsGroupCommand.cs

```diff
@@ -0,0 +1,134 @@
+using LT.DigitalOffice.ReactionService.Models.Dto.Requests.ReactionsGroup;
+using LT.DigitalOffice.ReactionService.Validation.ReactionsGroup.Interfaces;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ReactionService.Business.Commands.ReactionsGroup;
+
+public class CreateReactionsGroupCommand : ICreateReactionsGroupCommand
+{
+  private readonly ICreateReactionsGroupRequestValidator _validator;
+  private readonly IResponseCreator _responseCreator;
+  private readonly IAccessValidator _accessValidator;
+  private readonly IReactionsGroupRepository _reactionsGroupRepository;
+  private readonly IImageService _imageService;
+  private readonly IDbReactionsGroupMapper _reactionsGroupMapper;
+  private readonly IDbReactionMapper _reactionMapper;
+  private readonly IHttpContextAccessor _httpContextAccessor;
+  private readonly IImageCompressHelper _imageCompressHelper;
+  private readonly IImageResizeHelper _imageResizeHelper;
+
+  public CreateReactionsGroupCommand(
+    ICreateReactionsGroupRequestValidator validator,
+    IResponseCreator responseCreator,
+    IAccessValidator accessValidator,
+    IReactionsGroupRepository reactionsGroupRepository,
+    IImageService imageService,
+    IDbReactionsGroupMapper reactionsGroupMapper,
+    IDbReactionMapper reactionMapper,
+    IHttpContextAccessor httpContextAccessor,
+    IImageCompressHelper imageCompressHelper,
+    IImageResizeHelper imageResizeHelper)
+  {
+    _validator = validator;
+    _responseCreator = responseCreator;
+    _accessValidator = accessValidator;
+    _reactionsGroupRepository = reactionsGroupRepository;
+    _imageService = imageService;
+    _reactionsGroupMapper = reactionsGroupMapper;
+    _reactionMapper = reactionMapper;
+    _httpContextAccessor = httpContextAccessor;
+    _imageCompressHelper = imageCompressHelper;
+    _imageResizeHelper = imageResizeHelper;
+  }
+
+  public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateReactionsGroupRequest request)
+  {
+    if (!await _accessValidator.IsAdminAsync(_httpContextAccessor.HttpContext.GetUserId()))
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.Forbidden);
+    }
+
+    ValidationResult validationResult = await _validator.ValidateAsync(request);
+
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(
+        HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(x => x.ErrorMessage).ToList());
+    }
+
+    OperationResultResponse<Guid?> response = new();
+
+    List<DbReaction> dbReactions = new();
+
+    foreach (CreateReactionRequest reaction in request.Reactions)
+    {
+      (bool isResizeSuccess, string imageContent, reaction.Image.Extension) = await _imageResizeHelper.ResizeAsync(reaction.Image.Content, reaction.Image.Extension, 24);
```

Comment:
:84 :92 :98 lines are too long. Set enter after '=' for example.

Reply:
Done